File: org.opencompare/play-app/test/LoadPCMs.java
import model.Database;
import org.junit.Ignore;
import org.opencompare.api.java.Cell;
import org.opencompare.api.java.PCM;
import org.opencompare.api.java.PCMContainer;
import org.opencompare.api.java.Product;
import org.opencompare.api.java.impl.io.KMFJSONLoader;
import org.opencompare.formalizer.extractor.CellContentInterpreter;
import org.junit.Test;

import java.io.File;
import java.io.FileNotFoundException;
import java.io.FilenameFilter;
import java.io.IOException;
import java.util.List;

/**
 * Created by gbecan on 12/12/14.
 */
public class LoadPCMs {

    @Test
    public void testLoadWikipediaPCMs() throws IOException {
        String path = "../dataset-wikipedia/output/formalized/model";
        File dir = new File(path);

        KMFJSONLoader loader = new KMFJSONLoader();

        for (File file : dir.listFiles(new FilenameFilter() {
            @Override
            public boolean accept(File dir, String name) {
                return name.endsWith(".pcm");
            }
        })) {
            List<PCMContainer> pcmContainers = loader.load(file);
            PCMContainer pcmContainer = pcmContainers.get(0);
            PCM pcm = pcmContainer.getPcm();
            pcm.setName(pcm.getName().replaceAll("_", " "));

            if (pcm.isValid()) {
                Database.INSTANCE.create(pcmContainer);
            }
        }

    }

}

File: org.opencompare/play-app/test/LoadPCMs.scala
import java.io.{File, FilenameFilter}
import java.util.List

import models.Database
import org.junit.Test
import org.opencompare.api.java.impl.io.KMFJSONLoader

/**
  * Created by gbecan on 12/12/14.
  */
class LoadPCMs {

  @Test
  def testLoadWikipediaPCMs() {
    val path = "../dataset-wikipedia/output/formalized/model"
    val dir = new File(path)

    val loader = new KMFJSONLoader()

    for (file <- dir.listFiles.filter(_.getName.endsWith(".pcm"))) {
      // Load PCM
      val pcmContainers = loader.load(file)
      val pcmContainer = pcmContainers.get(0)
      val pcm = pcmContainer.getPcm()
      val metadata = pcmContainer.getMetadata()

      // Set source and license
      val hyphenIndex = pcm.getName.indexOf("-")
      val pageName = if (hyphenIndex >= 0) {
        pcm.getName.substring(0, hyphenIndex - 1)
      } else {
        pcm.getName
      }

      val source = "https://en.wikipedia.org/wiki/" + pageName
      metadata.setSource(source)
      metadata.setLicense("Creative Commons Attribution-ShareAlike 3.0 Unported")

      // Remove underscores in name
      pcm.setName(pcm.getName().replaceAll("_", " "))

      // Add to database if valid
      if (pcm.isValid()) {
        Database.create(pcmContainer)
      }
    }

  }

}
